xtask/src/commands/test.rs

```diff
@@ -19,6 +29,10 @@ pub struct Test {
     /// Do start the federation demo (without docker).
     #[structopt(long, conflicts_with = "no-demo")]
     with_demo: bool,
+
+    /// Start compiling the tests while federation demo is booting up.
+    #[structopt(long)]
```

Comment:
Make it conflicts with `with-demo`.

Reply:
you mean with no-demo ? this flag is only useful when with-demo is passed

Reply:
Right!! huh, sorry I was mistaking!!

So yes it is conflicting with `no-demo` BUT ALSO in the code you need to not start the bg process if `with_demo` is false.

Reply:
I can't think of a reason why people would use with_demo and not pre_compile tbh

Reply:
1. On my machine `npm install` took 3 sec:

```
Executed in    2.70 secs    fish           external
   usr time    3.91 secs  370.00 micros    3.91 secs
   sys time    0.28 secs   55.00 micros    0.28 secs
```

2. If `--with-demo` implies `--pre-compile` then you need to explain it in the CLI properly.

> Anyway. My point is that a cli parameter should be explicit. --with-demo should only mean "run the federation demo in background" and not "run the federation demo in background and start another background process to optimize your time". I prefer to have that under a separate parameter like --pre-compile.

Now, that being said. It's okay to disagree. If you do want `--with-demo` to imply pre-compiling, then change the flag `--pre-compile` to `--no-pre-compile` and document on `--with-demo` that it does run a pre-compilation process in background that can be disabled with `--no-pre-compile`. (just to be explicit)

Reply:
```
xtask-test 0.1.0-prealpha.1
Run tests for Router

USAGE:
    xtask test [FLAGS]

FLAGS:
    -h, --help              Prints help information
        --no-demo           Do not start federation demo (deprecated, this is the default now)
        --no-pre-compile    Do not start compiling the tests while federation demo is booting up. Requires --with-demo
    -V, --version           Prints version information
        --with-demo         Do start the federation demo (without docker). Xtask will pre compile the first test run
                            while federation-demo is booting up. If you want to disable pre_compile, use the --no-pre-
                            compile flag
```

wdyt ?